kobuki_driver/src/driver/core_sensors.cpp

```diff
@@ -45,7 +45,7 @@ bool CoreSensors::serialise(ecl::PushAndPop<unsigned char> & byteStream)
 }
 bool CoreSensors::deserialise(ecl::PushAndPop<unsigned char> & byteStream)
 {
-  if (byteStream.size() < length+2)
+  if (byteStream.size() < (unsigned int)(length+2))
```

Comment:
Needs some different casting (+2 looks like the problem). size() is unsigned int, length is unsigned char. Needs a static_cast.

Reply:
`ecl::PushAndPop()::size()` is returning an unsigned int, and `length` is an unsigned char.   I think the warning comes about because `2` is an int, and adding an int to an unsigned char promotes it to a signed int.  I changed the `2` to a `2U`, and the warning is gone.